src/test/java/duke/task/TaskListTest.java

```diff
@@ -0,0 +1,216 @@
+                            + "Got it. I've added this task:\n" + "[T][ ] buy bread\n"
+                            + "Now you have 1 tasks in the list\n"
+                            + "________________________________________\n",
+                    outContent.toString());
+            fail(); // the test should not reach this line
+        } catch (DukeException e) {
+            assertEquals("OOPS!!! The description of a todo cannot be empty.", e.getMessage());
+        }
+        restoreStreams();
+    }
+
+    @Test
+    public void addTask_addEventHalloween_exceptionThrownNoDescription() throws DukeException, IOException {
+        setUpStreams();
+        TaskList taskList = new TaskList();
+        UserInput testInput = new UserInput("E", "event", "","" , false);
+        try {
+            taskList.addTask(testInput);
+            assertEquals("________________________________________\n"
+                            + "Got it. I've added this task:\n" + "[E][ ] halloween (at: Sun)\n"
+                            + "Now you have 1 tasks in the list\n"
+                            + "________________________________________\n",
+                    outContent.toString());
+            fail(); // the test should not reach this line
+        } catch (DukeException e) {
+            assertEquals("OOPS!!! The description of a event cannot be empty.", e.getMessage());
+        }
+        restoreStreams();
+    }
+
+    @Test
+    public void addTask_addEventHalloween_exceptionThrownNoTime() throws DukeException, IOException {
+        setUpStreams();
+        TaskList taskList = new TaskList();
+        UserInput testInput = new UserInput("E", "event", "halloween ","" , false);
+        try {
+            taskList.addTask(testInput);
+            assertEquals("________________________________________\n"
+                            + "Got it. I've added this task:\n" + "[E][ ] halloween (at: Sun)\n"
+                            + "Now you have 1 tasks in the list\n"
+                            + "________________________________________\n",
+                    outContent.toString());
+            fail(); // the test should not reach this line
+        } catch (DukeException e) {
+            assertEquals("OOPS!!! Remember to set a time.", e.getMessage());
+        }
+        restoreStreams();
+    }
+
+    @Test
+    public void addTask_addDeadlineProject_exceptionThrownNoDescription() throws DukeException, IOException {
+        setUpStreams();
+        TaskList taskList = new TaskList();
+        UserInput testInput = new UserInput("D", "deadline", "","" , false);
+        try {
+            taskList.addTask(testInput);
+            assertEquals("________________________________________\n"
+                            + "Got it. I've added this task:\n" + "D][ ] proj (by: 2022-01-31)\n"
+                            + "Now you have 1 tasks in the list\n"
+                            + "________________________________________\n",
```

Comment:
Maybe this could become a static constant?

Reply:
Yeah, sounds good.